src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -1198,5 +1208,406 @@ static void verifySymbolInfo(IMethodSymbol methodSymbol, PublicNullableAnnotatio
+            Assert.Equal(PublicNullableAnnotation.NotAnnotated, symbol.NullableAnnotation);
+        }
+
+        [Fact]
+        public void GetDeclaredSymbol_ForLoop()
+        {
+            var source = @"
+class C
+{
+    void M(object? o1, object o2)
+    {
+        for (var o3 = o1; false; ) {}
+        for (var o4 = o1 ?? o2; false; ) {}
+        for (var o5 = o1, o6 = o2; false; ) {}
+    }
+}
+";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (8,14): error CS0819: Implicitly-typed variables cannot have multiple declarators
+                //         for (var o5 = o1, o6 = o2; false; ) {}
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedVariableMultipleDeclarator, "var o5 = o1, o6 = o2").WithLocation(8, 14));
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var declarations = root.DescendantNodes().OfType<VariableDeclaratorSyntax>().ToList();
+
+            assertAnnotation(declarations[0], PublicNullableAnnotation.Annotated);
+            assertAnnotation(declarations[1], PublicNullableAnnotation.NotAnnotated);
+            assertAnnotation(declarations[2], PublicNullableAnnotation.Annotated);
+            assertAnnotation(declarations[3], PublicNullableAnnotation.NotAnnotated);
+
+            void assertAnnotation(VariableDeclaratorSyntax variable, PublicNullableAnnotation expectedAnnotation)
+            {
+                var symbol = (ILocalSymbol)model.GetDeclaredSymbol(variable);
+                Assert.Equal(expectedAnnotation, symbol.NullableAnnotation);
+            }
+        }
+
+        [Fact]
+        public void GetDeclaredSymbol_OutVariable()
+        {
+            var source = @"
+class C
+{
+    void Out(out object? o1, out object o2) => throw null!;
+    void M()
+    {
+        Out(out var o1, out var o2);
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var declarations = root.DescendantNodes().OfType<SingleVariableDesignationSyntax>().ToList();
+            assertAnnotation(declarations[0], PublicNullableAnnotation.Annotated);
+            assertAnnotation(declarations[1], PublicNullableAnnotation.NotAnnotated);
+
+
+            void assertAnnotation(SingleVariableDesignationSyntax variable, PublicNullableAnnotation expectedAnnotation)
+            {
+                var symbol = (ILocalSymbol)model.GetDeclaredSymbol(variable);
+                Assert.Equal(expectedAnnotation, symbol.NullableAnnotation);
+            }
+        }
+
+        [Fact]
+        public void GetDeclaredSymbol_Switch()
```

Comment:
>GetDeclaredSymbol_Switch [](start = 20, length = 24)

consider adding a switch expression variant in test

Reply:
There is one :)

---
In reply to: [305060750](https://github.com/dotnet/roslyn/pull/37322#discussion_r305060750) [](ancestors = 305060750)